pkg/querier/queryrange/limits.go

```diff
@@ -58,15 +59,14 @@ func (l limits) QuerySplitDuration(user string) time.Duration {
 }
 
 // WithDefaults will construct a Limits with a default value for QuerySplitDuration when no overrides are present.
-func WithDefaultLimits(l Limits, conf queryrangebase.Config) Limits {
+func WithDefaultLimits(l Limits) Limits {
 	res := limits{
 		Limits:    l,
 		overrides: true,
 	}
 
-	if conf.SplitQueriesByInterval != 0 {
-		res.splitDuration = conf.SplitQueriesByInterval
-	}
+	// Set as the default split by interval value
+	res.splitDuration = l.QuerySplitDurationDefault()
```

Comment:
We should be able to remove `res.splitDuration` now, since it'll be automatically covered in the underlying `Limits` impl. Previously we needed to support `tenant override > tenant default > global default`, but we've removed `global default` and we already handle the first two in the underlying implementation: https://github.com/grafana/loki/blob/main/pkg/validation/limits.go#L530-L538

Reply:
Oh yeah, +1 to this as well.

Reply:
@owen-d I don't think we can remove `res.splitDuration` because of function WithSplitByLimits:
https://github.com/grafana/loki/blob/d35fcf361824a3ac360a21c9d0e75803ad0c6b06/pkg/querier/queryrange/limits.go#L82-L87
used to force a 24-hour in the query range middleware of the `NewSeriesTripperware` and `NewLabelsTripperware`